packages/treasury/src/vaultManager.js

```diff
@@ -182,10 +184,14 @@ export function makeVaultManager(
     updateState: updateTime =>
       chargeAllVaults(updateTime, poolIncrementSeat).catch(_ => {}),
     fail: reason => {
-      zcf.shutdownWithFailure(`Unable to continue without a timer: ${reason}`);
+      zcf.shutdownWithFailure(
+        assert.error(X`Unable to continue without a timer: ${reason}`),
```

Comment:
This looks surprising to me. When I read it, I see an `assert.error()`, which I expect to terminate operation, which is expected to return a value that will be passed to `zcf.shutdownWithFailure()`. Why is the latter able to gain control after the `assert.error()` call?

Reply:
Hi @Chris-Hibbert , I think you're thinking of `assert.fail` which both creates the error and throws it. `assert.error` has the same inputs and creates the same error, but returns it rather than throwing it.